src/user/user.controller.ts

```diff
@@ -43,16 +44,24 @@ export class UserController {
     if (!req.user || !req.body.email) {
       throw new BadRequestException('Invalid user data.');
     }
-    return this.userService.findByEmail(req.body.email);
+    const user = await this.userService.findByEmail(req.body.email);
```

Comment:
Move this logic to the service class

If the user is not found, it should throw an exception to return a 404 status code, instead of a custom error message

Reply:
I thought of doing that at first, but then I thought it would make more sense to acknowledge that the user still exists, and that its account has been deleted. In any case, let me know if I should still throw the NotFoundException and I'll do it